api/transport/outbound.go

```diff
@@ -64,6 +64,16 @@ type OnewayOutbound interface {
 	CallOneway(ctx context.Context, request *Request) (Ack, error)
 }
 
+// StreamOutbound is a transport that knows how to send stream requests for
+// procedure calls.
+type StreamOutbound interface {
+	Outbound
+
+	// CallStream creates a stream connection based on the metadata in the
+	// "Request" if the Request.Body is set, this should error.
+	CallStream(ctx context.Context, requestMeta *RequestMeta) (ClientStream, error)
```

Comment:
Offline discussion with @abhinav: Going to make a StreamRequest object that embeds a RequestMeta (so we have the ability to add extra Stream-Specific attributes here (just in case)

Reply:
A specific example of this would be distinguishing the "Create Connection" timeout of a request versus "Lifetime" timeout of the connection as a whole.